packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -32,15 +38,26 @@ export default class WebpackConfigGenerator {
     d('Config mode:', this.mode);
   }
 
-  resolveConfig(config: Configuration | string): Configuration {
-    if (typeof config === 'string') {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires, import/no-dynamic-require, global-require
-      return require(path.resolve(this.projectDir, config)) as Configuration;
-    }
+  async resolveConfig(config: Configuration | ConfigurationFactory | string): Promise<Configuration> {
+    const rawConfig =
+      typeof config === 'string'
+        ? // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
+          (require(path.resolve(this.projectDir, config)) as Configuration | ConfigurationFactory)
```

Comment:
It would be nice if this were ESM compatible, so that the config file here could be an ESM module (potentially w/ top-level await). I think @MarshallOfSound mentioned a module that will do the work of figuring out whether a file is CommonJS or ESM and "just do the right thing" i.e. `require()` or `import()`? Or maybe just `import()` is enough here?